### PR TITLE
convert EnvVariableFeatureFlag to an injected dependency

### LINKED ISSUE #21139: update EnvVariableFeatureFlags usages
The `EnvVariableFeatureFlags` class is currently being instantiated via code similar to:
```java
class Example {
  private final boolean logConnectorMessages = new EnvVariableFeatureFlags().logConnectorMessages();
}
```
This makes it difficult to migrate to the new [feature-flag implementation](#21091) due to the new client expecting to be injected as a dependency where necessary. 

This story is to update the usage of the `EnvVariableFeatureFlags` class to be dependency injected so that it can be swapped out easily with the new feature-flag work.

### PR DESCRIPTION
**Pull request by colesnodgrass**
## What
- fix #21139
- made it easier to change the feature-flag implementation by injecting the dependency instead of instantiating within the various classes modified in this PR
- does not modify integrations (will do as a separate PR)

## How
- replace `new EnvVariableFeatureFlags()` calls with injected `FeatureFlag` dependency

### CODE REVIEW

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Review by gosusnp — APPROVED** [review]

File: airbyte-commons-worker/src/test/java/io/airbyte/workers/general/ReplicationWorkerPerformanceTest.java [hunk 1]
```diff
@@ -65,7 +66,7 @@ public class ReplicationWorkerPerformanceTest {
   public void executeOneSync() throws InterruptedException {
     final var perSource = new LimitedAirbyteSource();
     final var perDestination = new EmptyAirbyteDestination();
-    final var messageTracker = new AirbyteMessageTracker();
+    final var messageTracker = new AirbyteMessageTracker(new EnvVariableFeatureFlags());
```
Comment (benmoriceau) [inline airbyte-commons-worker/src/test/java/io/airbyte/workers/general/ReplicationWorkerPerformanceTest.java:69]: Nit in the test should we use mock? The test might be affected by the env of where the test are run.

**Review by benmoriceau — APPROVED** [review]
on Nit comment on the test. Apart from that, I merged the server before your commit it is more FYI since this shouldn't impact you change.

File: airbyte-commons-worker/src/test/java/io/airbyte/workers/general/ReplicationWorkerPerformanceTest.java (hunk 1 above)
Comment (colesnodgrass) [reply airbyte-commons-worker/src/test/java/io/airbyte/workers/general/ReplicationWorkerPerformanceTest.java:69]: The tests should use a mock, and when I replace this with the new feature-flag client I'll add mocks where necessary.  

The code (and the tests) were and are already referencing the _real_ environment variables.  So the changes in this PR shouldn't have any unexpected side-effects that didn't already exist.

**Pull request merged**